public/js/user.js

```diff
@@ -356,6 +369,14 @@ apos.define('apostrophe-workflow', {
       }
       var i = 0;
       return async.eachSeries(ids, function(id, callback) {
+        if (self.skipAllRelated && (leadId !== id)) {
+          i++;
+          return setImmediate(callback);
+        }
+        if (self.commitAllRelated && (leadId !== id)) {
+          i++;
+          return self.commitSimilarly(id, callback);
```

Comment:
Is it possible both conditionals could be true, then it wouldn't reach this? Maybe they should be chained `else if` style.